### PR TITLE
Update CI to support microservice architecture

### LINKED ISSUE #45: Update CI to support new microservice architecture
Our current GHA workflow publishes a single image to our container registry. We need to update this workflow to publish geocml-desktop, geocml-server, geocml-postgres, and geocml-task-scheduler (separately) to our registry.

### PR DESCRIPTION
**Pull request by capsulecorplab**
closes #45 following pattern from https://github.com/docker/build-push-action/issues/561#issuecomment-1054596444

### CODE REVIEW

**Comment by capsulecorplab** [comment]
Images are building as of https://github.com/capsulecorplab/geocml-base-deployment/actions/runs/3774866663, but they get published under the container registry associated with the repo https://github.com/capsulecorplab/geocml-base-deployment/pkgs/container/geocml-base-deployment/versions

**Comment by capsulecorplab** [comment]
created a PR on my forked repo from the same branch to trigger/test CI build https://github.com/capsulecorplab/geocml-base-deployment/pull/1

File: .github/workflows/main.yml
Comment (TristanDamron) [inline .github/workflows/main.yml:11]: We don't have a dev branch on the repo, so I think we can remove this

File: .github/workflows/main.yml
Comment (TristanDamron) [inline .github/workflows/main.yml:22]: Changes to `devResources` and `docker-compose.yml` shouldn't be ignored--any changes here should trigger a new build

**Review by TristanDamron — CHANGES_REQUESTED** [review]
I think this should work! I left a couple of review comments for you, but once those are fixed we can merge this and see if it passes CI 🤞

File: .github/workflows/main.yml
Comment (capsulecorplab) [reply .github/workflows/main.yml]: resolved as per https://github.com/geoCML/geocml-base-deployment/pull/61/commits/4475f517d0f5e1704246ce2cb096ebf2cbeae8e1

File: .github/workflows/main.yml
Comment (capsulecorplab) [reply .github/workflows/main.yml]: resolved as per https://github.com/geoCML/geocml-base-deployment/pull/61/commits/dbe275dc9b60dee44789c1c8c1d77b76432f376b

**Pull request merged**
